src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +125,1497 @@ static ColliderCylinderInitType1 sCylinderInit = {
+            sSmallStumpSkullKid.skullkid = THIS;
+            this->actor.destroy = NULL;
+            this->actor.draw = NULL;
+            this->actor.update = EnSkj_OcarinaMinigameShortStumpUpdate;
+            this->actor.flags &= ~5;
+            this->actor.flags |= 0;
+            Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, thisx, ACTORCAT_PROP);
+            this->actor.focus.pos.x = 1230.0f;
+            this->actor.focus.pos.y = -90.0f;
+            this->actor.focus.pos.z = 450.0f;
+            this->actionFunc = EnSkj_SetupWaitForOcarina;
+            break;
+
+        default:
+            this->actor.params = type;
+            if (((this->actor.params != 0) && (this->actor.params != 1)) && (this->actor.params != 2)) {
+                if (gSaveContext.inventory.items[gItemSlots[ITEM_POCKET_EGG]] < ITEM_SAW) {
+                    Actor_Kill(&this->actor);
+                    return;
+                }
+            }
+
+            EnSkj_SetNaviId(this);
+            SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gSkullKidSkel, &gSkullKidPlayFluteAnim, this->jointTable,
+                               this->morphTable, 19);
+            if ((type >= 0) && (type < 3)) {
+                this->actor.flags &= ~5;
+                this->actor.flags |= 9;
+                Actor_ChangeCategory(globalCtx, &globalCtx->actorCtx, &this->actor, ACTORCAT_NPC);
+            }
+
+            if ((type < 0) || (type >= 7)) {
+                this->actor.flags &= ~0x02000000;
+            }
+
+            if ((type > 0) && (type < 3)) {
+                this->actor.targetMode = 7;
+                this->posCopy = this->actor.world.pos;
+                sOcarinaMinigameSkullKids[type - 1].unk0 = 1;
+                sOcarinaMinigameSkullKids[type - 1].skullkid = this;
+                this->minigameState = 0;
+                this->alpha = 0;
+                EnSkj_OcarinaGameSetupWaitForPlayer(this);
+            } else {
+                this->alpha = 255;
+                EnSkj_SetupResetFight(this);
+            }
+
+            this->actor.colChkInfo.damageTable = &sDamageTable;
+            this->actor.colChkInfo.health = 10;
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinderType1(globalCtx, &this->collider, &this->actor, &D_80B01678);
+            ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 40.0f);
+            Actor_SetScale(thisx, 0.01f);
+            this->textId = 0;
+            this->actor.textId = 0;
+            this->multiuseTimer = 0;
+            this->backfilpFlag = 0;
+            this->needlesToShoot = 3;
+            this->hitsUntilDodge = 3;
+            this->actor.speedXZ = 0.0f;
+            this->actor.velocity.y = 0.0f;
+            this->actor.gravity = -1.0f;
+            EnSkj_CalculateCenter(this);
+            globalCtx2 = globalCtx;
+            player = ((Player*)globalCtx2->actorCtx.actorLists[ACTORCAT_PLAYER].head);
```

Comment:
does this actually improve things? setting globalctx down here doesnt seem right at all